python/tests/lis/test_load.py

```diff
@@ -0,0 +1,622 @@
+        assert f1.reel_header().prev_reel_name  == ' '*8
+        assert f1.reel_trailer().next_reel_name == 'Reel5678'
+
+        # LF 2 is partitined correctly and belongs to Reel 2 and Tape 2
+        assert f2.header().file_name  == 'LISTST.002'
+        assert f2.trailer().file_name == 'LISTST.002'
+        assert f2.tape_header().prev_tape_name  == 'Tape0001'
+        assert f2.tape_trailer().next_tape_name == ' '*8
+        assert f2.reel_header().prev_reel_name  == 'Reel1234'
+        assert f2.reel_trailer().next_reel_name == ' '*8
+
+def test_partitioning_missing_many_delimiters(tmpdir, merge_lis_prs, assert_log):
+    # A file with missing delimiter:
+    #
+    # file.lis
+    # |
+    # |->Reel 1
+    # |  |-> Tape 1
+    # |      |-> Logical File 1
+    # |->Reel 2
+    #    |-> Tape 2
+    #        |-> Logical File 2
+    #
+    fpath = os.path.join(str(tmpdir), 'missing_many_delimiters.dlis')
+    content = [
+        # Missing first RHLR
+        # Missing first THLR
+        'data/lis/records/FHLR-1.lis.part',
+        'data/lis/records/FTLR-1.lis.part',
+        'data/lis/records/TTLR-1.lis.part',
+        # Missing non-last (in this case the first) RTLR
+        'data/lis/records/RHLR-2.lis.part',
+        # Missing non-first (in this case the last) THLR
+        'data/lis/records/FHLR-2.lis.part',
+        'data/lis/records/FTLR-2.lis.part',
+        # Missing last TTLR
+        # Missing last RTLR
+    ]
+    merge_lis_prs(fpath, content)
+
+    with lis.load(fpath) as (f1, f2, *tail):
+        assert len(tail) == 0
+
+        # LF 1 is partitined correctly and belongs to Reel 1 and Tape 1
+        assert f1.header().file_name  == 'LISTST.001'
+        assert f1.trailer().file_name == 'LISTST.001'
+        assert f1.tape_header()                 == None
+        assert f1.tape_trailer().next_tape_name == 'Tape0002'
+        assert f1.reel_header()                 == None
+        assert f1.reel_trailer()                == None
+
+        # LF 2 is partitined correctly and belongs to Reel 2 and Tape 2
+        assert f2.header().file_name  == 'LISTST.002'
+        assert f2.trailer().file_name == 'LISTST.002'
+        assert f2.tape_header()                 == None
+        assert f2.tape_trailer()                == None
+        assert f2.reel_header().prev_reel_name  == 'Reel1234'
+        assert f2.reel_trailer()                == None
+
+def test_partitioning_nodelimiters(tmpdir, merge_lis_prs, assert_log):
+    # A file that misses all delimiters:
+    #
+    # file.lis
+    # |
+    # |->Reel 1
+    # |  |-> Tape 1
+    # |      |-> Logical File 1
+    # |->Reel 2
+    #    |-> Tape 2
+    #        |-> Logical File 2
+    #
+    # This is now essentially just a file with 2 Logical Files without tape or
+    # reel information. I.e. all structure is lost
+    fpath = os.path.join(str(tmpdir), 'nodelimiters.dlis')
+    content = [
+        'data/lis/records/FHLR-1.lis.part',
+        'data/lis/records/FTLR-1.lis.part',
```

Comment:
Note: FTLR also might be a delimiter...
According to what is written on top of the 2.2.2.2 section, FTLR "if present"...
Customer subset on the other hand doesn't state it to be optional.